packages/notifi-react-card/lib/hooks/useIntercomChat.ts

```diff
@@ -52,6 +52,32 @@ export const useIntercomChat = ({
   const [isScrolling, setIsScrolling] = useState<boolean | null>();
   const { client } = useNotifiClientContext();
 
+  useEffect(() => {
+    const intervalId = setInterval(function () {
+      client
+        .getConversationMessages({
+          first: MESSAGES_NUMBER,
+          getConversationMessagesInput: { conversationId },
+        })
+        .then((response) => {
+          if (Array.isArray(response.nodes)) {
+            const nodes = response.nodes;
+            if (nodes[0].id === chatMessages[0].id) {
+              return;
+            } else {
+              nodes.forEach((node, index) => {
+                if (node.id === chatMessages[0].id) {
+                  const newMessages = nodes.slice(0, index);
+                  setChatMessages([...newMessages, ...chatMessages]);
+                }
+              });
+            }
```

Comment:
it's probably better to always add all the messages and dedupe on `id`, sorted by message created time.

This logic seems reliant on all new messages being "newer" than the last known newest message, which feels like it could run into edge cases with client-side insert of sendMessage results.

Reply:
I am adding the chatMessages as dependency here to avoid the edge case for sending messages, but yeah, it's still insecure. Let me redo the logic as you suggested 👍